src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -3799,6 +3805,159 @@ private bool SelfDescribingEvents
             }
         }
 
+#if NATIVEAOT
+        // If EventSource feature is enabled, default EventSources need to be initialized for NativeAOT
+        // In CoreCLR, this is done via a call from the runtime as part of coreclr_initialize
+#pragma warning disable CA2255
+        [ModuleInitializer]
+#pragma warning restore CA2255
+#endif
+        internal static void InitializeDefaultEventSources()
+        {
+            if(!EventSource.IsSupported)
+            {
+                return;
+            }
+
+// NOTE: this define is being used inconsistently. Most places mean just EventPipe support, but then a few places use
+// it to mean other aspects of tracing such as these EventSources.
+#if FEATURE_PERFTRACING
```

Comment:
Curious question. Why is this `#if #endif` required if all it does is assign some values that are actually discarded and not used?

Reply:
Its for the side effects. Reading the field forces the static constructor to run if it hasn't already. That in turn creates the two EventSource objects and stores them in static fields.

Reply:
Ok I see. Thanks for clarifying!